include/math_utils.h

```diff
@@ -151,6 +167,14 @@ int32_t clamp_to_int32(const T val)
 	return static_cast<int32_t>(std::clamp(val, min_val, max_val));
 }
 
+template <typename T>
+uint32_t clamp_to_uint32(const T val)
+{
+	constexpr auto min_val = static_cast<T>(0);
+	constexpr auto max_val = static_cast<T>(UINT32_MAX);
+	return static_cast<uint32_t>(std::clamp(val, min_val, max_val));
+}
+
```

Comment:
If you want to cherry-pick commit: ac1ccbc02d0809c3023bb6272da1e62db408d6f2 , this makes all the clampers `constexpr` and adds `static_assert`s to check if `val`'s type doesn't need clamping (in which case, `val`'s type range already is fully bounded within the clamping `T` range).  

I think it's a worthwhile check, because it can catch assumptions about the type being sent in (it actually caught a couple cases in my original unit tests :-) ).

Reply:
OK, cherry-picked - but could you explain the change in [bios.cpp](https://github.com/dosbox-staging/dosbox-staging/commit/8f6c8957ddaa89de09ec006b353c64afaa8c42be#diff-338d73fa3c6d7d35a8bb444b85dcbca91bcaab256c1fbb905d5d3d4c33ac4e52)? Why do we need this `static_cast<int64_t>`?

Reply:
On 32-bit systems, the output from `MEM_TotalPages()` is uint32_t, so many of the subsequent casts aren't needed.

![2023-02-20_08-54](https://user-images.githubusercontent.com/1557255/220164582-4290053c-7494-4a26-bd4d-d8181956a6b5.png)

So the use of int64_t was more a work-around (to lift it up to 64-bit just like when compiling on 64-bit archs), which let me keep the clamp back down to 32-bit. 

But really, MEM_TotalPages is never going to return anything close to billions of pages (let alone 64-bit's worth) - a better fix is to make `MEM_TotalPages()` return a `uint32_t`; then most of those casts and clamps would disappear here too.

Reply:
@kcgen I have changed return types of two functions. Please check the last commit.